framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,15 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
shouldnt this be put once in an ApplicationHelper?

Reply:
I'm for `Yii::ENCODING` constant or just leaving `UTF-8` string in ternary check. This approach is already used in multiple places in core. Symfony uses `UTF-8` as "magic string", and Laravel don't even care about different encodings and have hardcoded `UTF-8` in multiple places. It is not worth complicating this only for extremely edge cases.

Reply:
Leaving a string there sounds OK to me.